test/cpp/interop/client_helper.cc

```diff
@@ -88,11 +88,13 @@ std::shared_ptr<Channel> CreateChannelForTestCase(
 
   std::shared_ptr<CallCredentials> creds;
   if (test_case == "compute_engine_creds") {
-    GPR_ASSERT(FLAGS_use_tls);
+    GPR_ASSERT(FLAGS_use_tls ||
+               FLAGS_custom_credentials_type == "google_default_credentials");
```

Comment:
This is a little hard to follow, because when we're using "google default credentials", we set `creds = GoogleComputeEngineCredentials();` but then don't actually use those call `creds` (if I'm correct).

Can we only set `creds = GoogleComputeEngineCredentials();` in the case that we're using TLS, and leave creds null if we're using GCD?

Reply:
That's a good point. There is no need to set `creds` when using "google default credentials". The code will be revised.